model-optimizer/extensions/middle/RandomUniformReplacer.py

```diff
@@ -0,0 +1,64 @@
+"""
+ Copyright (C) 2018-2020 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+import numpy as np
+
+from typing import Dict
+
+from mo.front.tf.graph_utils import create_op_with_const_inputs
+from mo.graph.graph import Graph, Node, rename_nodes
+from mo.middle.replacement import MiddleReplacementPattern
+from mo.ops.broadcast import Broadcast
+
+
+class RandomUniformReplacer(MiddleReplacementPattern):
+    """
+    Replaces RandomUniform operation with Broadcast of ones in sub-graph:
+
+    ShapeOf ---> RandomUniform ---> Mul
+
+    """
+
+    enabled = True
+
+    @staticmethod
+    def pattern():
+        return dict(
+            nodes=[
+                ('shape', dict(kind='op', op='ShapeOf')),
+                ('shape_data', dict(kind='data')),
+                ('random_uniform', dict(kind='op', op='RandomUniform')),
+                ('random_uniform_data', dict(kind='data')),
+                ('mul', dict(kind='op', op='Mul'))
+            ],
+            edges=[
+                ('shape', 'shape_data'),
+                ('shape_data', 'random_uniform'),
+                ('random_uniform', 'random_uniform_data'),
+                ('random_uniform_data', 'mul', {'in': 0})
+            ]
+        )
+
+    @staticmethod
+    def replace_pattern(graph: Graph, match: Dict[str, Node]):
+        node = match['random_uniform']
+        node_name = node.soft_get('name', node.id)
+        data_type = node.in_port(0).get_data_type()
```

Comment:
it is not propagated yet. it should be taken from the other mul input. please match it as constant.